### ISSUE TITLE
pandas 2.0.0 error using "oed_fields" in analysis settings 

### ISSUE THREAD

**Issue by sambles** [OasisLMF/OasisLMF#1253]
## Issue Description
Using `oed_fields` 
```
    "gul_summaries": [ 
        { 
            "oed_fields": ["LocNumber"],
            "aalcalc": true,
            "eltcalc": true,
            "id": 1,
            "lec_output": true,
            "leccalc": { 
                "full_uncertainty_aep": true,
                "full_uncertainty_oep": true,
                "return_period_file": true
            }
        } 
    ],
```

Causes the following error with pandas==2.0.0 
```
Traceback (most recent call last):
  File "/home/sam/repos/venv/dev/bin/oasislmf", line 33, in <module>
    sys.exit(load_entry_point('oasislmf==1.27.2', 'console_scripts', 'oasislmf')())
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/cli/root.py", line 54, in main
    sys.exit(RootCmd().run())
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/cli/root.py", line 42, in run
    return super(OasisBaseCommand, self).run(args=args)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/argparsetree/cmd.py", line 159, in run
    return cmd_cls(sub_command_name).run(args)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/argparsetree/cmd.py", line 159, in run
    return cmd_cls(sub_command_name).run(args)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/argparsetree/cmd.py", line 161, in run
    return self.action(args) or 0
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/cli/command.py", line 142, in action
    manager_method(**_kwargs)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/utils/log.py", line 122, in wrapper
    result = func(*args, **kwargs)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/manager.py", line 90, in interface
    return computation_cls(**kwargs).run()
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/computation/run/model.py", line 80, in run
    cmd[0](**cmd[1]).run()
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/computation/generate/losses.py", line 600, in run
    analysis_settings = GenerateLossesDir.run(self)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/computation/generate/losses.py", line 284, in run
    generate_summaryxref_files(
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/utils/log.py", line 122, in wrapper
    result = func(*args, **kwargs)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/preparation/summaries.py", line 568, in generate_summaryxref_files
    gul_summaryxref_df, gul_summary_desc = get_summary_xref_df(
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/preparation/summaries.py", line 489, in get_summary_xref_df
    ) = group_by_oed(cols_group_by, map_df, exposure_df, id_set_index, accounts_df)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/oasislmf/preparation/summaries.py", line 188, in group_by_oed
    summary_group_df = summary_map_df.loc[:, set(mapped_cols).union(tiv_cols)]
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/pandas/core/indexing.py", line 1091, in __getitem__
    check_dict_or_set_indexers(key)
  File "/home/sam/repos/venv/dev/lib/python3.10/site-packages/pandas/core/indexing.py", line 2611, in check_dict_or_set_indexers
    raise TypeError(
TypeError: Passing a set as an indexer is not supported. Use a list instead.

```

**Comment by mtazzari** [comment]
I've encountered and fixed the same issue here: #1252

**Comment by mtazzari** [comment]
if `tiv_cols` is already a set, something like `...loc[:, list(tiv_cols.union(mapped_cols))]` should work.
